### PR TITLE
init a collection of bioinformatics tools

### PR DESCRIPTION
**Pull request by jbedo**
###### Motivation for this change
Add some commonly used bioinformatics tools.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/biology/platypus/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{stdenv, fetchFromGitHub, python27, htslib, zlib, makeWrapper}:
+
+let python = python27.withPackages (ps: with ps; [ cython ]);
+
+in stdenv.mkDerivation rec {
+  name = "platypus-${version}";
+  version = "git-2017-03-07";
```
Comment (rycee) [inline pkgs/applications/science/biology/platypus/default.nix]: This should be `name = "platypus-unstable-2017-03-07";`. See https://nixos.org/nixpkgs/manual/#sec-package-naming.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -17004,12 +17006,20 @@ with pkgs;
...
+
+  platypus = callPackage ../applications/science/biology/platypus/default.nix { };
+
   plink = callPackage ../applications/science/biology/plink/default.nix { };
 
   plink-ng = callPackage ../applications/science/biology/plink-ng/default.nix { };
 
   samtools = callPackage ../applications/science/biology/samtools/default.nix { };
 
+  SnpEff = callPackage ../applications/science/biology/SnpEff/default.nix { };
```
Comment (rycee) [inline pkgs/top-level/all-packages.nix]: I would suggest using lowercase letters here. Not a hard requirement but when possible it is preferred.

File: pkgs/applications/science/biology/platypus/default.nix (hunk 1 above)
Comment (jbedo) [inline pkgs/applications/science/biology/platypus/default.nix]: Thanks for the feedback, I have fixed the naming issues.

**Comment by rycee** [comment]
Rebased into master in commits 1ac7978, 076e513, 4bb50fd, 7f214e6, and 661048c. Note, I made some minor tweaks, for example in the commit messages as well as the license for varscan. Please look through the packages to verify that you agree with my changes.

Many thanks for your contribution!

**Pull request closed without merging**